### PR TITLE
Resolved task #499

### PR DESCRIPTION
**Pull request by Beniamin-Dolega**
Renamed SyncToRemoteStatus ==> Relation to Remote
Renamed Relation ==> SyncToRemoteStatus aditionally making it a top-level enum

### CODE REVIEW

File: backendApi/src/main/java/com/virtuslab/gitmachete/backend/api/RelationToRemote.java
Comment (PawelLipski) [inline backendApi/src/main/java/com/virtuslab/gitmachete/backend/api/RelationToRemote.java:10]: Remove this `TODO` as this issue will be resolved when this PR gets merged

File: backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java
Comment (PawelLipski) [inline backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java:641]: Rename the method adequately

File: backendImpl/src/test/java/com/virtuslab/gitmachete/backend/integration/ParentInferenceIntegrationTestSuite.java
Comment (PawelLipski) [inline backendImpl/src/test/java/com/virtuslab/gitmachete/backend/integration/ParentInferenceIntegrationTestSuite.java:28]: Pls add a `TODO (#753): un-ignore this suite`

Comment (PawelLipski) [reply backendImpl/src/test/java/com/virtuslab/gitmachete/backend/integration/ParentInferenceIntegrationTestSuite.java:28]: Your next task :)

File: backendImpl/src/test/java/com/virtuslab/gitmachete/backend/integration/StatusAndDiscoverIntegrationTestSuite.java
Comment (PawelLipski) [inline backendImpl/src/test/java/com/virtuslab/gitmachete/backend/integration/StatusAndDiscoverIntegrationTestSuite.java:38]: Ditto, pls add a TODO

File: backendImpl/src/test/java/com/virtuslab/gitmachete/backend/integration/StatusAndDiscoverIntegrationTestSuite.java
Comment (PawelLipski) [inline backendImpl/src/test/java/com/virtuslab/gitmachete/backend/integration/StatusAndDiscoverIntegrationTestSuite.java:212]: Rename the local variable

Comment (PawelLipski) [reply backendImpl/src/test/java/com/virtuslab/gitmachete/backend/integration/StatusAndDiscoverIntegrationTestSuite.java:212]: Btw, `val` (short for _value_) means two things:
1. the type is automatically guessed based on what you assign to the variable (you don't need to specify it yourself)
2. the variable is **final** (can't be ever reassigned), so it's not really a variable but right more of a constant

File: backendImpl/src/test/java/com/virtuslab/gitmachete/backend/unit/GitMacheteRepository_deriveSyncToRemoteStatusToRemoteUnitTestSuite.java
Comment (PawelLipski) [inline backendImpl/src/test/java/com/virtuslab/gitmachete/backend/unit/GitMacheteRepository_deriveSyncToRemoteStatusToRemoteUnitTestSuite.java:22]: Whoops too much stuff in this name

File: backendImpl/src/test/java/com/virtuslab/gitmachete/backend/unit/GitMacheteRepository_deriveSyncToRemoteStatusToRemoteUnitTestSuite.java
Comment (PawelLipski) [inline backendImpl/src/test/java/com/virtuslab/gitmachete/backend/unit/GitMacheteRepository_deriveSyncToRemoteStatusToRemoteUnitTestSuite.java:33]: Hmm is this method still called `deriveSyncToRemoteStatus` even though it returns a `RelationToRemote`?

Comment (PawelLipski) [reply backendImpl/src/test/java/com/virtuslab/gitmachete/backend/unit/GitMacheteRepository_deriveSyncToRemoteStatusToRemoteUnitTestSuite.java:33]: Tip: Ctrl+Shift+Alt+N to locate any **symbol**, including class, method etc.

Comment (PawelLipski) [reply backendImpl/src/test/java/com/virtuslab/gitmachete/backend/unit/GitMacheteRepository_deriveSyncToRemoteStatusToRemoteUnitTestSuite.java:33]: (while Ctrl+N just looks for any **class**)

File: docs/features.md
Comment (PawelLipski) [inline docs/features.md:13]: Whoops

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePushBranchAction.java
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePushBranchAction.java:50]: Rename the method adequately

File: frontendResourceBundles/src/main/resources/GitMacheteBundle.properties
Comment (PawelLipski) [inline frontendResourceBundles/src/main/resources/GitMacheteBundle.properties:163]: Let's leave `relation` here, as it was (that's just a message for a user, we can be a little bit informal here, as compared to the naming we use in the code itself)

File: frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/DemoGitMacheteRepositorySnapshot.java
Comment (PawelLipski) [inline frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/DemoGitMacheteRepositorySnapshot.java:92]: Rename the method, you're now getting _relation of status_ after the renames, not _status of relation_ ;)

File: frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/cell/BranchOrCommitCellRendererComponent.java
Comment (PawelLipski) [inline frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/cell/BranchOrCommitCellRendererComponent.java:261]: Rename the param

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ISyncToRemoteStatusDependentAction.java
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ISyncToRemoteStatusDependentAction.java:69]: Rename the var

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ISyncToRemoteStatusDependentAction.java
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ISyncToRemoteStatusDependentAction.java:67]: Rename the val

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ISyncToRemoteStatusDependentAction.java
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ISyncToRemoteStatusDependentAction.java:45]: As mentioned, rename this method

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ISyncToRemoteStatusDependentAction.java
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ISyncToRemoteStatusDependentAction.java:36]: Nitpick:
```suggestion
   * This method provides a list of {@link SyncToRemoteStatus}es for which the action should be ENABLED.
```

File: backendImpl/src/test/java/com/virtuslab/gitmachete/backend/unit/GitMacheteRepository_deriveSyncToRemoteStatusToRemoteUnitTestSuite.java
Comment (Beniamin-Dolega) [reply backendImpl/src/test/java/com/virtuslab/gitmachete/backend/unit/GitMacheteRepository_deriveSyncToRemoteStatusToRemoteUnitTestSuite.java:22]: Should it be called `GitMacheteRepository_deriveSyncToRemoteStatusUnitTestSuite` or `GitMacheteRepository_deriveRelationToRemoteUnitTestSuite` now?

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePushBranchAction.java
Comment (Beniamin-Dolega) [reply frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePushBranchAction.java:50]: Should it be named `getEligibleStatuses` now?

File: backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java
Comment (PawelLipski) [inline backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java:404]: Rename the val

File: backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java
Comment (PawelLipski) [inline backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java:460]: Ditto

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePushBranchAction.java
Comment (PawelLipski) [reply frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePushBranchAction.java]: Yes (pls don't resolve a thread if you wait for a response coz the reviewers are then more likely to overlook)

File: backendImpl/src/test/java/com/virtuslab/gitmachete/backend/unit/GitMacheteRepository_deriveSyncToRemoteStatusToRemoteUnitTestSuite.java
Comment (PawelLipski) [reply backendImpl/src/test/java/com/virtuslab/gitmachete/backend/unit/GitMacheteRepository_deriveSyncToRemoteStatusToRemoteUnitTestSuite.java]: `GitMacheteRepository_deriveRelationToRemoteUnitTestSuite` (current name is okay)

File: docs/features.md
Comment (PawelLipski) [inline docs/features.md:191]: Revert

File: docs/features.md
Comment (PawelLipski) [inline docs/features.md:220]: Revert

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ISyncToRemoteStatusDependentAction.java
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ISyncToRemoteStatusDependentAction.java:67]: Lower case, as it's a local val, not e.g. a class

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ISyncToRemoteStatusDependentAction.java
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ISyncToRemoteStatusDependentAction.java:70]:
```suggestion
    val isStatusEligible = getEligibleStatuses().contains(status);
```

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ISyncToRemoteStatusDependentAction.java
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ISyncToRemoteStatusDependentAction.java:83]: TBH I'd remove the `.relation` part from all the `action.GitMachete.ISyncToRemoteStatusDependentAction.` property keys

Comment (PawelLipski) [reply frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ISyncToRemoteStatusDependentAction.java:83]: You need to do the change both where the keys are referenced (so, in this Java file) and where they're declared (i.e. in `frontendResourceBundles/src/main/resources/GitMacheteBundle.properties`)

File: frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/cell/BranchOrCommitCellRendererComponent.java
Comment (PawelLipski) [inline frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/cell/BranchOrCommitCellRendererComponent.java:178]: Nitpick:
```suggestion
      String relationToRemoteLabel = getRelationToRemoteBasedLabel(relationToRemote);
```

File: frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/cell/BranchOrCommitCellRendererComponent.java
Comment (PawelLipski) [inline frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/cell/BranchOrCommitCellRendererComponent.java:261]: Rename the param

**Review by PawelLipski — APPROVED** [review]
Good job 🎉 as soon as CircleCI passes, you can Squash & Merge this pull request via button here on GitHub

**Pull request merged**
